src/structarray.jl

```diff
@@ -455,9 +459,9 @@ combine_style_types(s::BroadcastStyle) = s
 
 Base.@pure cst(::Type{SA}) where SA = combine_style_types(array_types(SA).parameters...)
 
-BroadcastStyle(::Type{SA}) where SA<:StructArray = StructArrayStyle{typeof(cst(SA))}()
+BroadcastStyle(::Type{SA}) where SA<:StructArray = StructArrayStyle{typeof(cst(SA)),ndims(SA)}()
 
-Base.similar(bc::Broadcasted{StructArrayStyle{S}}, ::Type{ElType}) where {S<:DefaultArrayStyle,N,ElType} =
+Base.similar(bc::Broadcasted{<:StructArrayStyle{S}}, ::Type{ElType}) where {S<:DefaultArrayStyle,N,ElType} =
```

Comment:
```suggestion
Base.similar(bc::Broadcasted{StructArrayStyle{S,N}}, ::Type{ElType}) where {S<:DefaultArrayStyle,N,ElType} =
```

I'm a bit confused as to why the version above worked at all, with an unmatched type parameter.

Reply:
The `similar` is extended in the latest version